web/html/src/manager/maintenance/edit/schedule-edit.js

```diff
@@ -0,0 +1,129 @@
+/* eslint-disable */
+'use strict';
+
+import React, {useState, useEffect, forwardRef, useImperativeHandle} from "react";
+import {Check} from "components/input/Check";
+import {Form} from "components/input/Form";
+import {Text} from "components/input/Text";
+import {Radio} from "components/input/Radio";
+import {Button} from "components/buttons";
+import {Combobox} from "components/combobox";
+
+type ScheduleEditProps = {
+    isEdit: boolean,
+    schedule?: {
+        id: number,
+        name: string,
+        type: 'SINGLE' | 'MULTI',
+        calendarName: string,
+    },
+    calendarNames: Array<Map<number, string>>,
+    onEdit: () => void,
+};
+
+const MaintenanceScheduleEdit = forwardRef((props: ScheduleEditProps, ref) => {
+    const [model, setModel] = useState({
+        name: "",
+        type: "SINGLE",
+        calendarName: "",
+        strategy: false
+    });
+    const [calendarAdded, setCalendarAdded] = useState(false);
+    const [selectedCalendar, setSelectedCalendar] = useState(0);
+
+    useEffect(() => {
+        if(props.isEdit) {
+            setModel({...model,
+                name: props.schedule.name,
+                type: props.schedule.type,
+                calendarName: props.schedule.calendarName
+            });
+            setCalendarAdded(!!props.schedule.calendarName);
```

Comment:
Haha if I do it with `props.schedule.calendarName ? true : false` my IDE highlights the code and tells me it can be simplified to `!!props.schedule.calendarName`. It was just to tempting to press the button :D

Reply:
Your IDE is too smart for us :sweat_smile: